internal/component/pyroscope/receive_http/receive_http.go

```diff
@@ -0,0 +1,182 @@
+package receive_http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/grafana/alloy/internal/component/pyroscope/write"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/grafana/alloy/internal/component"
+	fnet "github.com/grafana/alloy/internal/component/common/net"
+	"github.com/grafana/alloy/internal/component/pyroscope"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "pyroscope.receive_http",
+		Stability: featuregate.StabilityGenerallyAvailable,
+		Args:      Arguments{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Server    *fnet.ServerConfig     `alloy:",squash"`
+	ForwardTo []pyroscope.Appendable `alloy:"forward_to,attr"`
+}
+
+// SetToDefault implements syntax.Defaulter.
+func (a *Arguments) SetToDefault() {
+	*a = Arguments{
+		Server: fnet.DefaultServerConfig(),
+	}
+}
+
+type Component struct {
+	opts        component.Options
+	server      *fnet.TargetServer
+	appendables []pyroscope.Appendable
+	mut         sync.Mutex
+}
+
+func New(opts component.Options, args Arguments) (*Component, error) {
+	c := &Component{
+		opts:        opts,
+		appendables: args.ForwardTo,
+	}
+
+	if err := c.Update(args); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (c *Component) Run(ctx context.Context) error {
+	defer func() {
+		c.mut.Lock()
+		defer c.mut.Unlock()
+		c.shutdownServer()
+	}()
+
+	<-ctx.Done()
+	level.Info(c.opts.Logger).Log("msg", "terminating due to context done")
+	return nil
+}
+
+func (c *Component) Update(args component.Arguments) error {
+	newArgs := args.(Arguments)
+
+	c.mut.Lock()
+	defer c.mut.Unlock()
+
+	c.appendables = newArgs.ForwardTo
+
+	if newArgs.Server == nil {
+		newArgs.Server = fnet.DefaultServerConfig()
```

Comment:
In general prefer the ToDefault interface instead of setting in the Update unless there is a reason we cant.

Reply:
From the `Defaulter` [interface](https://github.com/grafana/alloy/blob/main/syntax/types.go#L35-L36) definition:
> SetToDefault will not be called on types which are squashed into the parent struct using `alloy:",squash"

Since we're defining `Arguments` as:
```golang
type Arguments struct {
  Server    *fnet.ServerConfig     `alloy:",squash"`
  ...
}
```
Is our `SetToDefault` implementation going to work? It's used this way in loki's [code base](https://github.com/grafana/alloy/blob/main/internal/component/loki/source/api/api.go#L31-L44) as well 🤔

Reply:
Forgive me you are correct.

Reply:
I updated the documentation but kept the `SetToDefault` implementation as a safeguard against potential future changes in squash behavior. Let me know if you'd prefer to remove it.